src/lib/pages/evm-contract-details/index.tsx

```diff
@@ -0,0 +1,181 @@
+import { track } from "@amplitude/analytics-browser";
+import {
+  Divider,
+  Stack,
+  TabList,
+  TabPanel,
+  TabPanels,
+  Tabs,
+} from "@chakra-ui/react";
+import { useRouter } from "next/router";
+import { useCallback, useEffect } from "react";
+
+import { AmpEvent, trackUseTab } from "lib/amplitude";
+import {
+  useConvertHexAddress,
+  useEvmConfig,
+  useInternalNavigate,
+} from "lib/app-provider";
+import { AssetsSection } from "lib/components/asset";
+import { CustomTab } from "lib/components/CustomTab";
+import { Loading } from "lib/components/Loading";
+import PageContainer from "lib/components/PageContainer";
+import { CelatoneSeo } from "lib/components/Seo";
+import { ErrorFetching, InvalidState } from "lib/components/state";
+import { UserDocsLink } from "lib/components/UserDocsLink";
+import { useBalanceInfos } from "lib/services/bank";
+import {
+  useEvmCodesByAddress,
+  useEvmContractInfoSequencer,
+} from "lib/services/evm";
+import type { HexAddr20 } from "lib/types";
+import { is0x, isHexWalletAddress, truncate } from "lib/utils";
+
+import { EvmContractDetailsBytecode } from "./components/EvmContractDetailsBytecode";
+import { EvmContractDetailsOverview } from "./components/EvmContractDetailsOverview";
+import { EvmContractDetailsTop } from "./components/EvmContractDetailsTop";
+import { TabIndex, zEvmContractDetailsQueryParams } from "./types";
+
+const InvalidContract = () => <InvalidState title="Contract does not exist" />;
+
+const tableHeaderId = "evmContractDetailsTab";
+
+interface EvmContractDetailsBodyProps {
+  contractAddress: HexAddr20;
+  tab: TabIndex;
+}
+
+const EvmContractDetailsBody = ({
+  contractAddress,
+  tab,
+}: EvmContractDetailsBodyProps) => {
+  const navigate = useInternalNavigate();
+  const { convertHexWalletAddress } = useConvertHexAddress();
+  const contractAddressBechAddr = convertHexWalletAddress(contractAddress);
+
+  const {
+    data: evmCodesByAddressData,
+    isLoading: isEvmCodesByAddressLoading,
+    isError: isEvmCodesByAddressError,
+  } = useEvmCodesByAddress(contractAddress);
+  const { data: evmContractInfoData, isLoading: isEvmContractInfoLoading } =
+    useEvmContractInfoSequencer(contractAddress);
+
+  const { totalData: totalAssets = 0 } = useBalanceInfos(
+    contractAddressBechAddr
+  );
+
+  const handleTabChange = useCallback(
+    (nextTab: TabIndex) => () => {
+      if (nextTab === tab) return;
+      trackUseTab(nextTab);
+      navigate({
+        pathname: "/evm-contracts/[contractAddress]/[tab]",
+        query: {
+          contractAddress,
+          tab: nextTab,
+        },
+        options: {
+          shallow: true,
+        },
+      });
+    },
+    [contractAddress, tab, navigate]
+  );
+
+  if (isEvmCodesByAddressLoading) return <Loading />;
+  if (
+    isEvmCodesByAddressError ||
```

Comment:
let's remove error for now. `!evmCodesByAddressData` should be enough and consistent with other pages